src/s3ql/fsck.py

```diff
@@ -1384,5 +1400,37 @@ def to_str(name):
     return str(name, encoding='utf-8', errors='replace')
 
 
+def verify_metadata_snapshots(backend, count=5):
+    '''Verify the last *count* metadata snapshots.'''
+
+    log.info('Verifying consistency of most recent metadata backups:')
+    backups = sorted(get_available_seq_nos(backend))
+
+    if not backups:
+        raise RuntimeError(
+            'No metadata backups found. This should not happen, please report a bug.'
+        )
+
+    for seq_no in backups[-count:]:
+        # De-serialize directly instead of using read_remote_params() to avoid
+        # exceptions on old filesystem revisions.
+        d = thaw_basic_mapping(backend['s3ql_params_%010x' % seq_no])
+        if d['revision'] != CURRENT_FS_REV:
+            break
```

Comment:
might `continue` instead of `break` be correct here? As far as I see, `backups` is sorted ascending so this code will not check any metadata when the first metadata to check is one with a different file system revision.

Reply:
Yes, thanks!